File: app/flytec/lib/features/aplications/models/identificacao_area_tratada.dart

class IdentificacaoAreaTratada {
  String? uf;
  String? cidade;
  String? localizacao;
  String? cultura;
  String? extensao;
  String? croquiArea;
  bool? isPdf;
  int? id;

  IdentificacaoAreaTratada({
    this.uf,
    this.cidade,
    this.localizacao,
    this.cultura,
    this.extensao,
    this.croquiArea,
    this.isPdf,
    this.id,
  });

  Map<String, dynamic> toMap() {
    return {
      'uf': uf,
      'cidade': cidade,
      'localizacao': localizacao,
      'cultura': cultura,
      'extensao': extensao,
      'croquiArea': croquiArea,
      'isPdf': isPdf! ? 1 : 0,
    };
  }

  factory IdentificacaoAreaTratada.fromJson(Map<String, dynamic>? json) {
    return IdentificacaoAreaTratada(
      uf: json?['uf'] ?? '',
      cidade: json?['cidade'] ?? '',
      localizacao: json?['localizacao'] ?? '',
      cultura: json?['cultura'] ?? '',
      extensao: json?['extensao'] ?? '',
      croquiArea: json?['croquiArea'],
      isPdf: json?['isPdf'] == 1,
      id: json?['id'] ?? 0,
    );
  }
}

File: backoffice/Domain/Entidades/Cadastros/IdentificacaoAreaTratada/IdentificacaoAreaTratada.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Domain.Entidades.Cadastros.IdentificacaoAreaTratada
{
    public class IdentificacaoAreaTratada
    {
        public int Id { get; set; }
        public string UF { get; set; }
        public string Cidade { get; set; }
        public string Localizacao { get; set; }
        public string Cultura { get; set; }
        public string Extensao { get; set; }
        public string CroquiArea { get; set; }
        public string? GravacaoArea { get; set; }
        public string? Marcadores { get; set; }

        [ForeignKey("Empresa")]
        public int? IdEmpresa { get; set; }

        [JsonIgnore]
        public virtual Empresa.Empresa? Empresa { get; set; }
    }
}
